### PR TITLE
Feed segment 0 its runtime input

### PR DESCRIPTION
**Pull request by zzw4257**
## The problem

A model compiled through Piper trains on zeros. Whatever is passed to `load_input` never reaches the graph, so the network sees an all-zero input on every step. On torch 2.10.0 this affects the shipped LLaMA debug model too.

## Why it happens

`split_gm_by_annotations` has to tell two kinds of placeholder apart: the real runtime inputs, and the parameters and module attributes Dynamo lifted into the graph signature. `_placeholder_is_runtime_input` used `meta["grapharg"]` as the marker of a lifted value.

Dynamo attaches that key to *every* placeholder while the backend is running and clears it once `torch.compile` returns. The real model input carried it as well, so it was classified as lifted, and segment 0 came out with `input_idxs=[]`.

Nothing downstream complains about an empty `input_idxs`. `_load_stage` zero-fills any slot that is neither a trainable parameter nor a matching const attr, and the FWD executor arm only substitutes real tensors for the indices listed in `input_idxs`. The input data is dropped in silence.

## Why it stayed invisible

The examples report iteration time, throughput and peak memory, and the loss does not reach the driver, so nothing in the printed output depends on the input values. Training converges to something, it just is not a function of the data.

It surfaced on an MLP with no biases: the output was exactly zero, and the MSE loss came out at `mean(labels^2)` to seven digits.

## What this PR changes

One clause in `_placeholder_is_runtime_input`:

```diff
-    if "self" in node.name or "grapharg" in node.meta:
+    if "self" in node.name:
```

The three checks above it already exclude parameters and anything with `requires_grad`, and the `"self"` name check still excludes lifted module attributes such as `l_self_freqs_cis` and `l_self_mask`, which have to stay graphargs so `load_const_attrs` can fill them.

After the change, segment 0 gets `input_idxs=[0]` for both the LLaMA debug model and a plain MLP.

## Verifying it

`test/test_runtime_inputs.py` asserts the classification from inside the backend call, while the meta key is still present; asserting it after `torch.compile` returns would pass with or without the fix. Reverting the one-line change turns both tests red.

## Context

First of ten for #15. It stands on its own, and it is also what makes a numerical comparison against a baseline possible at all — without it, two configurations agree because both are computing on zeros.

### CODE REVIEW

**Comment by stephanie-wang** [comment]
I had some difficulty understanding the PR description. Can you update to clarify the context and what this PR does?

**Comment by zzw4257** [comment]
You're right, sorry — that description started in the middle of the diagnosis and never said what was actually wrong. Rewritten.

The short version: a model compiled through Piper trains on zeros. `_placeholder_is_runtime_input` used `meta["grapharg"]` to spot placeholders Dynamo had lifted, but Dynamo puts that key on *every* placeholder while the backend is running, so the real model input was classified as lifted too and segment 0 got `input_idxs=[]`. `_load_stage` then zero-fills the slot and `load_input`'s data is dropped in silence. It reaches the shipped LLaMA debug model as well.

It stays invisible because the examples report time, throughput and peak memory, and the loss does not reach the driver, so nothing printed depends on the input values.

The fix is dropping one clause from the condition; the description now walks through why the remaining checks still cover the cases that clause was there for.

I have also rewritten #34 the same way, and will use that shape for the rest of the series — what is wrong or missing, how it lowers, what the PR deliberately leaves out, how to check it. Happy to adjust if you would rather they read differently.
